python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -478,3 +479,40 @@ def cleanup(self):
     @staticmethod
     def bootstrap_config(cluster_config):
         return bootstrap_aws(cluster_config)
+
+    @staticmethod
+    def fillout_available_node_types_resources(
+            cluster_config: Dict[str, Any]) -> Dict[str, Any]:
+        """Fills out missing "resources" field for available_node_types."""
```

Comment:
Can we deep copy `cluster_config` at the start of the function so that this operation is transactional? I'm worried about failures in the for loop. (Relying on the ec2 api/boto3 to handle edge cases stably feels a little sketchy).